### PR TITLE
Add raycast-bundlephobia extension

### PR DESCRIPTION
**Pull request by EvanZhouDev**
## Description

Use Bundlephobia in Raycast!

See the following data:

- Minified Size
- Minified + GZIP Size
- Download times for 3G and 4G
- Dependency List

## Screencast

For example, here's what you get if you run it with Vue.js:

raycast-bundlephobia-1

raycast-bundlephobia-2

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by EvanZhouDev** [comment]
Well, I guess another Bundlephobia extension already exists, but again, this is still a different concept, where you can directly check the package size. I might consider adding this into the other Bundlephobia extension actually...

**Comment by pernielsentikaer** [comment]
Let's hear the author, maybe this can superseed the other one? What do you think @dgrzedzielski

**Comment by pernielsentikaer** [comment]
@dgrzedzielski are you still around?

**Comment by dgrzedzielski** [comment]
Hey, sorry for the delay

I think it could replace mine, but what I would like to change is to make the font size of bundle sizes bigger / more highlighted.

File: extensions/raycast-bundlephobia/src/analyze-package.jsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import { popToRoot, Detail, showToast, Toast } from "@raycast/api";
+import fetch from "node-fetch";
+import { useState, useEffect } from "react";
+
+export default function AnalyzePackage(props) {
+  const { packageName } = props.arguments;
+
+  const [markdown, setMarkdown] = useState("Loading data from Bundlephobia...");
+
+  function formatBytes(bytes) {
```
Comment (dgrzedzielski) [inline extensions/raycast-bundlephobia/src/analyze-package.jsx:10]: This function could be extracted outside of the component

File: extensions/raycast-bundlephobia/src/analyze-package.jsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    } else if (bytes < 1024 * 1024 * 1024) {
+      return (bytes / (1024 * 1024)).toFixed(1) + "mB";
+    } else if (bytes < 1024 * 1024 * 1024 * 1024) {
+      return (bytes / (1024 * 1024 * 1024)).toFixed(1) + "gB";
+    } else {
+      return (bytes / (1024 * 1024 * 1024 * 1024)).toFixed(1) + " tB";
+    }
+  }
+
+  useEffect(() => {
```
Comment (dgrzedzielski) [inline extensions/raycast-bundlephobia/src/analyze-package.jsx:24]: I think too much stuff is happening in this useEffect, I would suggest to extract some related stuff to separate functions, like: fetching data, processing / calulating data, generating markdown

File: extensions/raycast-bundlephobia/src/analyze-package.jsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      } catch {
+        popToRoot();
+        showToast({
+          style: Toast.Style.Failure,
+          title: "Failed to fetch library.",
+        });
+      }
+    })();
+  });
+  return <Detail loading={markdown === "Loading data from Bundlephobia..."} markdown={markdown} />;
```
[the remark this answers is not in the document]
Comment (pernielsentikaer) [reply extensions/raycast-bundlephobia/src/analyze-package.jsx:107]: Yeah, that's an unusual way 🙂

**Comment by pernielsentikaer** [comment]
Are you still around @EvanZhouDev

**Pull request closed without merging**